storage/types.go

```diff
@@ -234,7 +234,7 @@ type databaseShard interface {
 	Close() error
 
 	// Tick performs any updates to ensure series drain their buffers and blocks are flushed, etc
-	Tick(softDeadline time.Duration) tickResult
+	Tick(softDeadline time.Duration, c context.Cancellable) tickResult
```

Comment:
Perhaps the cancellable should always come first (maybe second only to a context.Context) if we want to follow similar conventions?  Go linter now makes you always put the context as a first param of a method, so it seems kind of natural we would put a cancellable as first, and second only if there was already a context as the first arg.   Thoughts?